scripts/test.sh

```diff
@@ -120,7 +123,7 @@ function integration_pass {
 
 function e2e_pass {
   # e2e tests are running pre-build binary. Settings like --race,-cover,-cpu does not have any impact.
-  run_for_module "tests" go_test "./e2e/..." "keep_going" : -timeout="${TIMEOUT:-30m}" "${RUN_ARG[@]}" "$@"
+  run_for_module "tests" go_test "./e2e/..." "keep_going" : -timeout="${TIMEOUT:-30m}" "${RUN_ARG[@]}" "$@" || return $?
```

Comment:
> If not, test.sh just reports it fails but no error exit code.

Could you explain this? Isn't it a bug? The script should exit on error, because of `set -e`

Reply:
Thanks for the clarification.

The `exit 255 ` will still exit the script?

Reply:
Really good catch!! It works after updating your example like below. 

```
#!/usr/bin/env bash

set -euo pipefail

main() {
  command -v xx || return $? # should fail
  command -v ls
}

#echo $SHELLOPTS
if ! main; then
  echo oops
fi
echo $?

main
```